### PR TITLE
Add the doc on Secret customization

### PR DESCRIPTION
**Pull request by IwonaLanger**
**Description**

Changes proposed in this pull request:

- Add the document on how to customize the `sap-btp-manager` Secret
- Update `03-10-preconfigured-secret.md`
- Adjust `03-70-delete-bindings-and-instances.md`

**Related issue(s)**

See #901

### CODE REVIEW

File: docs/user/03-10-preconfigured-secret.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ The `sap-btp-manager` Secret provides the following credentials:
 
 > [!NOTE]
 > If you modify or delete the `sap-btp-manager` Secret, it is modified back to its previous settings or regenerated within up to 24 hours.
-> However, if the Secret is labeled with `kyma-project.io/skip-reconciliation: "true"`, the job skips the reconciliation for this Secret.
+> To stop the reconciliation, label the Secret with `kyma-project.io/skip-reconciliation: "true"`. For more information, see [Customize Credentials and Access](03-11-customize_secret.md).
```
Comment (jaroslaw-pieszka) [inline docs/user/03-10-preconfigured-secret.md:32]: There is implicit assumption that reader is aware that `reconciliation` == `being modified back or regenerated`.
It was already in the previous version, but still.

File: docs/user/03-11-customize_secret.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+To use your own credentials as the default and manage your cluster on your own, customize the `sap-btp-manager` Secret.
+
+## Context
+
+When you create SAP BTP, Kyma runtime, the `sap-btp-manager` Secret is automatically created as the default Secret managing the SAP BTP service operator's resources. 
+Any changes to the Secret are reverted, and the previous settings are restored within up to 24 hours.
+See [Preconfigured Credentials and Access](03-10-preconfigured-secret.md).
+
+To stop the reconciliation of the `sap-btp-manager` Secret, you must customize it. With the customized Secret, you can perform the following actions:
```
Comment (jaroslaw-pieszka) [inline docs/user/03-11-customize_secret.md:11]: I see this the other way around.
To customize you must stop the reconciliation.

File: docs/user/03-11-customize_secret.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Customize the Default Credentials and Access
+
+You can customize the `sap-btp-manager` Secret and manage your own SAP BTP Operator's default configuration.
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:3]: Does this mean "manage the default configuration of your own SAP BTP Operator", or "manage your own default configuration of the SAP BTP Operator"?
I assume the 2nd option, anyway I'd suggest phrasing it in an unambiguous way,

File: docs/user/03-11-customize_secret.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Context
+
+When you create SAP BTP, Kyma runtime, the `sap-btp-manager` Secret is automatically created as the default Secret managing the SAP BTP Operator's resources. 
+Any changes to the Secret are reverted, and the previous settings are restored within up to 24 hours.
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:8]: Because the word reconciliation comes up in line 11, I'd introduce it here:

```suggestion
Because of Kyma's automatic reconciliation, any changes to the Secret are reverted, and the previous settings are restored within up to 24 hours.
```
(maybe you can phrase it more elegantly :D)

File: docs/user/03-11-customize_secret.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+See [Preconfigured Credentials and Access](03-10-preconfigured-secret.md#credentials).
+
+To customize the `sap-btp-manager` Secret and prevent your changes from being reverted, you must stop the Secret's reconciliation.
+With the customized Secret, you can perform the following actions:
+
+* Manage your own SAP BTP Operator's default configuration
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:14]: like in line 3, I find this noun phrase ambiguous. Can you disentangle it a bit?

File: docs/user/03-11-customize_secret.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+* Migrate the instances you created outside of the Kyma environment to your Kyma cluster
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:15]: instances of what? Kyma instances? Application instances? Service instances?

File: docs/user/03-11-customize_secret.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+To customize the `sap-btp-manager` Secret, perform the following steps:
+
+* Label the Secret with `kyma-project.io/skip-reconciliation: 'true'`.
+* Provide the credentials of the Kyma instance you want to migrate or recreate.
+* Optionally, add the `management_namespace` parameter and provide the name of your custom management namespace.
+
+## Result
+
+Your customized `sap-btp-manager` Secret is now the default SAP BTP Operator's Secret. It generates the SAP BTP service operator's resources, as shown in the following diagram:
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:27]: preposition instead of genitive makes it easier to understand for me

```suggestion
Your customized `sap-btp-manager` Secret is now the default Secret of the SAP BTP Operator. It generates the SAP BTP service operator's resources, as shown in the following diagram:
```

File: docs/user/03-11-customize_secret.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+The reconciliation of the Secret stops and your changes are not reverted.
+
+> [!WARNING]
+> If you delete the customized `sap-btp-manager` Secret, the reconciliation starts again, and the preconfigured default `sap-btp-manager` Secret is recreated for your Kyma instance within 24 hours. See [Preconfigured Credentials and Access](./03-10-preconfigured-secret.md#credentials).
+
+## Next Steps
+
+With the customized default `sap-btp-manager` Secret, you can perform the following actions:
+
+* Migrate your instances created, for example, in Gardener, to the Kyma environment.
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:40]: same question as in line 15 :D

**Review by NHingerl — APPROVED** [review]

**Review by nataliasitko — DISMISSED** [review]

File: docs/user/03-11-customize_secret.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+To customize the `sap-btp-manager` Secret and prevent your changes from being reverted, you must stop the Secret's reconciliation.
+With the customized Secret, you can perform the following actions:
+
+* Manage your own default configuration of SAP BTP Operator
+* Migrate the service instances you created outside of the Kyma environment to your Kyma cluster
+
+## Procedure
+
+To customize the `sap-btp-manager` Secret, perform the following steps:
```
Comment (ralikio) [inline docs/user/03-11-customize_secret.md:22]: The world "steps" made me think that I need to conduct steps in specific order but it is more about remembering to do the following modification to the secret. Could you rephrase "... perform the following steps" to something like "... perform the following secret modifications"?

File: docs/user/03-11-customize_secret.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  tokenurl: {AUTH_URL}
+  cluster_id: {CLUSTER_ID}
+  credentials_namespace: {CREDENTIALS_NAMESPACE}
+```
+
+## Result
+
+Your customized `sap-btp-manager` Secret is now the default Secret of the SAP BTP Operator module. It generates the SAP BTP service operator's resources, as shown in the following diagram:
+
+![Customized module credentials](../assets/module_credentials_customized.drawio.svg)
```
Comment (ralikio) [inline docs/user/03-11-customize_secret.md:53]: Could we revert the direction of arrows between sap-btp-service-operator and sap-btp-manager secrets? Now it looks like sap-btp-manager points to values set in sap-btp-service-operator.

File: docs/user/03-10-preconfigured-secret.md [hunk 11]
```diff
@@ -43,12 +43,14 @@ The cluster ID represents a Kyma service instance created in a particular subacc
...
-By default, SAP BTP Operator has cluster-wide permissions. You cannot reconfigure the predefined settings.
+By default, SAP BTP Operator has cluster-wide permissions.
 
 The following parameters manage cluster access:
 
 | Parameter                     | Description                                                                                   |
 |-------------------------------|-----------------------------------------------------------------------------------------------|
 | **CLUSTER_ID**                | Generated when Kyma runtime is created.                                                       |
-| **MANAGEMENT_NAMESPACE**      | Always set to `kyma-system`.                                                |
+| **MANAGEMENT_NAMESPACE**      | By default, set to `kyma-system`.                                                |
```
Comment (ralikio) [inline docs/user/03-10-preconfigured-secret.md:53]: I am not sure that this config does and I cannot find it in out documentation. Maybe we could clarify it at least as a followup?

File: docs/user/03-11-customize_secret.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+* Migrate the service instances you created outside of the Kyma environment to your Kyma cluster
+
+## Procedure
+
+To customize the `sap-btp-manager` Secret, perform the following steps:
+
+* Label the Secret with `kyma-project.io/skip-reconciliation: 'true'`.
+* Provide the following credentials from your SAP Service Manager instance: **clientid**, **clientsecret**, **sm_url**, and **tokenurl**.
+* Optionally, provide your **cluster_id**. Otherwise, it is generated automatically.
+* Optionally, add the **credentials_namespace** parameter and provide the name of your custom namespace for Secrets with credentials to communicate with the SAP Service Manager.
```
Comment (ralikio) [inline docs/user/03-11-customize_secret.md:27]: After reading this sentence I understood that the custom namespace should already contain credentials to use. Is that correct understanding? If not, could you change to something like "...the name of your custom namespace where  sap-btp-operator-service Secret will be generated."?

File: docs/user/03-10-preconfigured-secret.md [hunk 13]
```diff
@@ -43,12 +43,14 @@ The cluster ID represents a Kyma service instance created in a particular subacc
...
 | **ALLOW_CLUSTER_ACCESS**      | You can use every namespace for your operations. The parameter is always set to `true`.<br>If you change it to `false`, the setting is automatically reverted. |
```
Comment (PK85) [inline docs/user/03-10-preconfigured-secret.md:54]: @IwonaLanger please add here RELEASE_NAMESPACE too. Value: "By default, set to `kyma-system`."

File: docs/user/03-11-customize_secret.md (hunk 12 above)
Comment (szwedm) [reply docs/user/03-11-customize_secret.md:27]: That's correct, we assume the namespace already exists and it could contain anything user wants there. `sap-btp-service-operator` Secret also exists at this point, we just move it to the other namespace selected by user, it should remain the default Secret for communication with SAP Service Manager.

File: docs/user/03-11-customize_secret.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+![Customized module credentials](../assets/module_credentials_customized.drawio.svg)
+
+The reconciliation of the Secret stops and your changes are not reverted.
+
+> [!WARNING]
+> If you delete the customized `sap-btp-manager` Secret, the reconciliation starts again, and the preconfigured default `sap-btp-manager` Secret is recreated for your Kyma instance within 24 hours. See [Preconfigured Credentials and Access](./03-10-preconfigured-secret.md#credentials).
+
+> [!NOTE]
+> If you created all service instances in your Kyma cluster from the customized `sap-btp-manager` Secret, you can delete the cluster even if those instances still exist.
+> The undeleted service instances do not block the deletion of the cluster.
```
Comment (jaroslaw-pieszka) [inline docs/user/03-11-customize_secret.md:62]: What do we mean by `undeleted`?  User can delete, then undelete the service instance? Or this is just not-deleted service instance?

File: docs/user/03-11-customize_secret.md (hunk 9 above)
Comment (jaroslaw-pieszka) [reply docs/user/03-11-customize_secret.md:22]: > To customize the `sap-btp-manager` Secret, perform the following secret modifications.
Sounds a bit awkward almost like 
> To modify the `sap-btp-manager` Secret, perform the following secret modifications.

On the other hand the order is not relevant provided we do all modifications during single edit operation (kubectl edit). 
But we are not sure how user will proceed so it is safer to impose order beginning with setting the label to prevent reconciliation.

So I would vote to leave it as it is.

File: docs/user/03-10-preconfigured-secret.md (hunk 11 above)
Comment (IwonaLanger) [reply docs/user/03-10-preconfigured-secret.md:53]: I'm not sure what to clarify here. Can you please clarify? This section is based on [this source](https://github.com/SAP/sap-btp-service-operator/blob/main/README.md#managing-access) and modified according to PK's input.

File: docs/user/03-11-customize_secret.md (hunk 9 above)
Comment (IwonaLanger) [reply docs/user/03-11-customize_secret.md:22]: I'll rephrase the sentence if it causes confusion, but mind the fact that we can have ordered steps for procedures when the order of performing instructions is essential and unordered steps for cases like this one when you must perform them all in any order.

File: docs/user/03-11-customize_secret.md (hunk 12 above)
Comment (IwonaLanger) [reply docs/user/03-11-customize_secret.md:27]: This is a rule. Using "will" is against the [content guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#tenses).

File: docs/user/03-11-customize_secret.md (hunk 10 above)
Comment (IwonaLanger) [reply docs/user/03-11-customize_secret.md:53]: I think that is correct as "It generates the SAP BTP service operator's resources."

File: docs/user/03-11-customize_secret.md (hunk 14 above)
Comment (IwonaLanger) [reply docs/user/03-11-customize_secret.md:62]: Didn't know that. Fixed.

File: docs/user/03-11-customize_secret.md (hunk 10 above)
Comment (jaroslaw-pieszka) [reply docs/user/03-11-customize_secret.md:53]: Values from `sap-btp-manager` are the source during reconciliation and `sap-btp-service-operator` is the target.

File: docs/user/03-11-customize_secret.md (hunk 9 above)
Comment (ralikio) [reply docs/user/03-11-customize_secret.md]: I understood that the document follows a [Task Topic](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/00-content-strategy.md#task-topics) structure. As I understand there would be only one step in this document: "create a secret" and this is just a description of that step and the order does not matter in that paragraph.

File: docs/user/03-11-customize_secret.md (hunk 12 above)
Comment (szwedm) [reply docs/user/03-11-customize_secret.md:27]: The **credentials_namespace** is the namespace for secrets, it does not concern only the `sap-btp-service-operator` secret. The sentence `...the name of your custom namespace where sap-btp-operator-service Secret will be generated."` does not hold the true meaning of the namespace. User can create `sap-btp-manager` secret with the setting already provided and there won't be any movement of the secrets.

File: docs/user/03-10-preconfigured-secret.md (hunk 11 above)
Comment (IwonaLanger) [reply docs/user/03-10-preconfigured-secret.md]: More info has been added.

File: docs/user/03-10-preconfigured-secret.md [hunk 15]
```diff
@@ -47,10 +47,13 @@ By default, SAP BTP Operator has cluster-wide permissions.
...
-| **MANAGEMENT_NAMESPACE**      | By default, set to `kyma-system`.                                                |
-| **ALLOW_CLUSTER_ACCESS**      | You can use every namespace for your operations. The parameter is always set to `true`.<br>If you change it to `false`, the setting is automatically reverted. |
+| Parameter                | Description                                                                                                                                                                              |
+|--------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **CLUSTER_ID**           | Generated when Kyma runtime is created.                                                                                                                                                  |
+| **MANAGEMENT_NAMESPACE** | Indicates the namespace for Secrets referenced in service instances and Secrets with the name containing a prefix of the service instance's namespace. By default, set to `kyma-system`. |
```
Comment (NHingerl) [inline docs/user/03-10-preconfigured-secret.md:53]:
```suggestion
| **MANAGEMENT_NAMESPACE** | Indicates the namespace for Secrets referenced in service instances and Secrets, with the name containing a prefix of the service instance's namespace. By default, set to `kyma-system`. |
```
I'm not sure the comma is grammatically necessary, but considering the sentence is pretty long, I think it might improve comprehension.

File: docs/user/03-11-customize_secret.md [hunk 16]
```diff
@@ -1,5 +1,8 @@
 # Customize the Default Credentials and Access
 
+> [!NOTE]
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:3]: we shouldn't document features that don't exist yet. 
https://wiki.one.int.sap/wiki/display/PSUA/UA-030

I understand that probably no one will care about following the product standard for our kyma-project.io docs, but I'm not sure what's good practice here.

File: docs/user/03-70-delete-bindings-and-instances.md [hunk 17]
```diff
@@ -4,18 +4,11 @@ Delete service bindings and service instances using Kyma dashboard or kubectl.
 
 ## Context
 
-You can only delete service instances or service bindings created in Kyma using Kyma dashboard or kubectl. You can't perform these operations using the SAP BTP cockpit.
+You can only delete service instances or service bindings created in Kyma using Kyma dashboard or kubectl. You can't perform these operations using the SAP BTP cockpit. To successfully delete a service instance, first delete its service bindings.
```
Comment (NHingerl) [inline docs/user/03-70-delete-bindings-and-instances.md:7]: do we need the word "successfully" here?

File: docs/user/03-70-delete-bindings-and-instances.md [hunk 18]
```diff
@@ -40,4 +33,14 @@ To delete a service instance, run:
...
 ```
-<!-- tabs:end -->
\ No newline at end of file
+<!-- tabs:end -->
+
+## Next Steps
+
+If you want to delete your Kyma cluster, you may have to first delete your service instances and bindings from it.
+
+You can't delete your Kyma cluster if any non-deleted service instances in it use the credentials from the SAP Service Manager resources created automatically, as described in [Preconfigured Credentials and Access](03-10-preconfigured-secret.md#credentials). In this case, the existing service instances block the cluster's deletion. Delete your service instances and bindings in Kyma dashboard before you attempt to delete the cluster from the SAP BTP cockpit.
```
Comment (NHingerl) [inline docs/user/03-70-delete-bindings-and-instances.md:42]: I'd reverse the order of the "if" sentence and of the "before" sentence:

```suggestion
If any non-deleted service instances in your Kyma cluster use the credentials from the SAP Service Manager resources created automatically, you can't delete your Kyma cluster (see [Preconfigured Credentials and Access](03-10-preconfigured-secret.md#credentials). In this case, the existing service instances block the cluster's deletion. Before you attempt to delete the cluster from the SAP BTP cockpit, delete your service instances and bindings in Kyma dashboard.
```

File: docs/user/03-70-delete-bindings-and-instances.md [hunk 19]
```diff
@@ -40,4 +33,14 @@ To delete a service instance, run:
...
+
+You can delete your Kyma cluster even if your service instances still exist, provided they all use credentials of SAP Service Manager service instances other than the one created automatically, as described in [Preconfigured Credentials and Access](03-10-preconfigured-secret.md#credentials). In this case, the non-deleted service instances do not block the cluster's deletion.
```
Comment (NHingerl) [inline docs/user/03-70-delete-bindings-and-instances.md:44]:
```suggestion
However, if the service instances use credentials of SAP Service Manager service instances other than the one created automatically, you can delete your Kyma cluster even if your service instances still exist (see [Preconfigured Credentials and Access](03-10-preconfigured-secret.md#credentials)). In this case, the non-deleted service instances do not block the cluster's deletion. 
```

**Review by NHingerl — COMMENTED** [review]
just minor suggestions

File: docs/user/03-70-delete-bindings-and-instances.md (hunk 19 above)
Comment (IwonaLanger) [reply docs/user/03-70-delete-bindings-and-instances.md:44]: Same here.

File: docs/user/03-70-delete-bindings-and-instances.md (hunk 17 above)
Comment (IwonaLanger) [reply docs/user/03-70-delete-bindings-and-instances.md:7]: Nope :)

File: docs/user/03-11-customize_secret.md (hunk 16 above)
Comment (IwonaLanger) [reply docs/user/03-11-customize_secret.md:3]: I already explained. Let's try this way for a while.

File: docs/user/03-70-delete-bindings-and-instances.md (hunk 18 above)
Comment (NHingerl) [reply docs/user/03-70-delete-bindings-and-instances.md:42]: If the link target is the same for both cases, why not extract it and put it into an introductory sentence?

Instead of:
"If you want to delete your Kyma cluster, you may have to first delete your service instances and bindings from it."

Maybe:
"Deleting the service instances and bindings may be a prerequisite for deleting your Kyma cluster. This depends on the way the credentials have been created (for details, see [Preconfigured Credentials and Access](03-10-preconfigured-secret.md#credentials))."

In this case, you can drop the links from the following paragraphs and put the condition first.


Thinking about it more - the content of the Next Steps comes after the reader presumably successfully deleted service instances and bindings (because that is _the_ goal of this document). The scope of _this_ document is not how to delete a Kyma cluster successfully, so are the detailed explanations really necessary here? I think they don't hurt, but where they actually should be (and maybe they are?) is in documentation about deleting a Kyma cluster. (...do we have documentation about deleting a Kyma cluster?).

File: docs/user/03-10-preconfigured-secret.md [hunk 20]
```diff
@@ -29,7 +29,7 @@ The `sap-btp-manager` Secret provides the following credentials:
```
Comment (NHingerl) [inline docs/user/03-10-preconfigured-secret.md:29]: Line 13, I'd put "by default" first:

"By default, the SAP BTP Operator module is installed together with:"

File: docs/user/03-70-delete-bindings-and-instances.md (hunk 19 above)
Comment (IwonaLanger) [reply docs/user/03-70-delete-bindings-and-instances.md:44]: Skipping "all" and moving the link changes the message.

File: docs/user/03-70-delete-bindings-and-instances.md (hunk 18 above)
Comment (IwonaLanger) [reply docs/user/03-70-delete-bindings-and-instances.md:42]: If you phrase it this way, it will be difficult to understand what exactly [Preconfigured Credentials and Access](https://github.com/kyma-project/btp-manager/pull/933/03-10-preconfigured-secret.md#credentials) explains. And that is the automatic creation of the SAP Service Manager service instance.
The style guide mentions the [exception for very long conditions](https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/0e1ae688897b4bc185aa17002bae8ef4.html?state=DRAFT&q=condition#logical-sequence), and I think that's exactly the case.

Comment (IwonaLanger) [reply docs/user/03-70-delete-bindings-and-instances.md:42]: Let me think about where to put this content. In this section, I only mention what is related to the module. Cluster deletion is a broader topic.

File: docs/user/03-11-customize_secret.md [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+
+> [!NOTE]
+> This feature is still being developed and will be available soon.
+
+You can customize the `sap-btp-manager` Secret and manage your own default configuration of the SAP BTP Operator module.
+
+## Context
+
+When you create SAP BTP, Kyma runtime, the `sap-btp-manager` Secret is automatically created as the default Secret managing the SAP BTP Operator's resources. 
+Because of Kyma's automatic reconciliation, any changes to the Secret are reverted, and the previous settings are restored within up to 24 hours.
```
Comment (NHingerl) [inline docs/user/03-11-customize_secret.md:11]:
```suggestion
Because of Kyma's automatic reconciliation, any changes to the Secret are reverted, and the previous settings are restored within 24 hours.
```
Do we need this?

**Review by NHingerl — COMMENTED** [review]
It might look nicer if the three boxes had the same size: 
image

Sorry I didn't notice earlier :(

File: docs/user/03-11-customize_secret.md (hunk 21 above)
Comment (IwonaLanger) [reply docs/user/03-11-customize_secret.md:11]: Not necessarily. Removed.

**Review by NHingerl — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]
